lib/src/_shared/DateTextField.tsx

```diff
@@ -77,7 +77,7 @@ const getError = (
 
 export interface DateTextFieldProps
   extends WithUtilsProps,
-    Omit<TextFieldProps, 'onError' | 'onChange' | 'value'> {
+    Omit<StandardTextFieldProps, 'onError' | 'onChange' | 'value'> {
```

Comment:
I don't know how this component is used but remember that this way the variants `outlined` and `filled` won't be accepted by typescript.

You need to dissect the union into it's base types, omit and then join them again by their variant which is pretty complicated. I think the easiest solution is to join with `{ value: never }` instead of using `Omit` but this will have worse error messages if somebody actually passes the `value` prop.

Reply:
@eps1lon I'm trying to get ts working first with all the code, then we can move forward with flexibility for other variants as an enhancement.  @dmtrKovalenko made a first pass but there is much I am uncovering as I turn tests to ts/tsx.  Trying to minimize changes in this PR but it is big already.